SWLOR.Game.Server/Feature/UsePerkFeat.cs

```diff
@@ -239,8 +239,11 @@ void CompleteActivation(string id, float abilityRecastDelay)
                 Activity.ClearBusy(activator);
 
                 // Moved during casting or activator died. Cancel the activation.
-                if (GetLocalInt(activator, id) == (int)ActivationStatus.Interrupted || GetCurrentHitPoints(activator) <= 0)
+                if (GetLocalInt(activator, id) == (int)ActivationStatus.Interrupted || GetCurrentHitPoints(activator) <= 0 || GetCurrentAction(activator) != ActionType.Invalid)
```

Comment:
This is going to interrupt casting if any action is being performed, including an automatic attack by the player if they happen to be in melee. Have you confirmed that queued abilities (i.e weapon abilities) are not impacted by this change?

Reply:
I was able to queue pistol abilities in testing without any difficulty.